BHoM_Adapter/HelperMethods/PrepareObjects.cs

```diff
@@ -0,0 +1,69 @@
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Base;
+using BH.oM.Base;
+using BH.oM.Structure.Elements;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace BH.Adapter
+{
+    public static partial class Modify
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        public static List<IObject> WrapNonBHoMObjects(IEnumerable<IObject> objects, AdapterConfig adapterConfig, string tag = "", Dictionary<string, object> pushConfig = null)
+        {
+            bool wrapNonBHoMObjects = adapterConfig.WrapNonBHoMObjects;
+            object wrapNonBHoMObjValue;
+
+            if (pushConfig != null && pushConfig.TryGetValue("WrapNonBHoMObjects", out wrapNonBHoMObjValue)) wrapNonBHoMObjects = (bool)wrapNonBHoMObjValue;
+
+
+            List<IObject> objectsToPush = objects.Select(x =>
+            {
+                if (x is BHoMObject)
+                    return ((BHoMObject)x).DeepClone(); 
+
+                if (wrapNonBHoMObjects)
+                    return new CustomObject() { CustomData = new Dictionary<string, object> { { "WrappedObject", x } } }; // Wraps non-IBHoMObject in a custom BHoMObject
+
+                // If none of the above applies, return the non-BHoMObject untouched
+                return x;
+            })
+            .ToList(); //ToList() necessary for the return collection to function properly for cloned objects
+
+            return objectsToPush;
+        }
+
+        public static List<IObject> CloneBHoMObjects(IEnumerable<IObject> objects, AdapterConfig adapterConfig)
+        {
+            if (adapterConfig.CloneBeforePush)
+                return objects.Select(x => x is BHoMObject ? ((BHoMObject)x).DeepClone() : x).ToList(); // Deep clone for immutability in the UI
```

Comment:
No need to check if BHoMObject here I think. Deepclone method should be working on any object

Reply:
Thanks, done!